tests/functional/TicketTask.php

```diff
@@ -683,4 +689,373 @@ public function testParentMetaSearchOptions()
         $this->integer($data['data']['totalcount'])->isEqualTo(1);
         $this->string($data['data']['rows'][0]['Ticket_1'][0]['name'])->isEqualTo('ticket title');
     }
+
+    /**
+     * Test that the user can update his own task
+     * Right : TicketTask::UPDATEMY
+     *
+     * @return void
+     */
+    public function testUpdateMyTask()
+    {
+        global $DB;
+
+        $this->login();
+        $uid = getItemByTypeName(User::class, TU_USER, true);
+
+        //Create profile
+        $profile = $this->createItem(Profile::class, [
+            'name' => 'testUpdateMyTask',
+            'interface' => 'central',
+        ]);
+
+        $profile_user = new Profile_User();
+        $profile_user->getFromDBByCrit([
+            'users_id' => $uid,
+        ]);
+
+        $DB->delete(
+            ProfileRight::getTable(),
+            [
+                'profiles_id' => $profile->getID(),
+                'name' => 'task'
+            ],
+        );
+
+        //Add the profile to the user
+        $DB->update(
+            Profile_User::getTable(),
+            ['profiles_id' => $profile->getID()],
+            ['users_id' => $uid],
+        );
```

Comment:
It could be simplified. Same remark for other right update operations of other tests.

```suggestion
        $uid = getItemByTypeName(User::class, TU_USER, true);
        $profile_id = getItemByTypeName(Profile::class, 'Super-Admin', true);
    
        $DB->update(
            'glpi_profilerights',
            ['rights' => 0],
            [
                'profiles_id'  => $profile_id,
                'name'         => 'task'
            ]
        );
```